### ISSUE TITLE
New mode for marking

### ISSUE THREAD

**Issue by klump** [HGustavs/LenaSYS#1940]
I would very much like a new mode for correcting student submissions. It should be fairly similar to the reviewing process for questions and answers on stackoverflow.
After entering the mode, it displays a submission which needs grading with all the required information (like the fist-hover-thingie). Then there are at least 3 buttons somewhere (if possible with keybindings): pass, fail, and next. If the assignment has more grading options, add more buttons...
The next button should leave the submission untouched.
If feedback to the student is required, a popup with an auto-focus text-area should be displayed.

Make lenasys more keyboard-friendly because using mouse sucks...

**Comment by b14davbo** [comment]
:+1: for Shortcuts. Sounds like a good idea :)

**Comment by HGustavs** [comment]
Well close to what we want....

This is what we want. 
* The current rollover window, but shown on click instead of on rollover.
* And the window auto-coses when the mark is clicked (same round buttons as in the marking view)
* If you either hit escape or an X top right in the window it closes.

The "next" button is problematic as there is no natural "next", the "next" to one person may not be the "next of another". 

In short: Cross or Escape to Close, same window as always but with round marking buttons window closes after click on a mark, X or escape key.

**Comment by klump** [comment]
From my perspective, the next assignment could be any submitted assignment without a final grade (pass or fail or whatever). Preferably of the I will see all ungraded submissions for one assignment first, then moving on to the next ungraded assignment. The order does not really matter, eventually I have to grade all the submissions...

**Comment by a14nikau** [comment]
Mail from Klump:

Dugga/DuggaSys/resulted.php ß This is the problem ;)
 
It is:
·         Slow to load
·         Slow to respond to the on hover events (loading and generating the pages)
·         Very slow to save and update the grade
·         And the Javascript seems blocking. If I report a grade, the whole site freezes I can move the mouse but it will not render the next hover grading page. Instead it takes at least 10 seconds (I have counted) to report the grade, update the grade and then shows the new grading window. Very much time spend waiting – it gets frustrating…

**Comment by HGustavs** [comment]
When the new logging system is in place we can make a better analysis of what the problem is.

Either the performance problem is in the database or in the javascript.

Also, it is a good idea to supply some test data that allow us to test and evaluate this.

The performance is a problem mainly when there is a course with many students (>100) and many assignments (>30).

**Comment by klump** [comment]
That is basically why I created the issue in the first place. I am currently grading in a course with 128 students and tons of assignments. Everything feels slow and sluggy. Also I only need to see one submission at a time, lenasys displays all of them which makes the browser slow. It is slow in chrome and firefox becomes unusable.

**Comment by a14nikau** [comment]
This issue might be addressed in the redesign suggestion in issue #2136
